<launch>
  <arg name="laser_offset_x" />
  <arg name="laser_offset_y" />
  <arg name="leave_meetpoint_min_dist" />
  <arg name="leave_meetpoint_max_dist" />

  <node name="gvg_mapper" pkg="gvg_mapper" type="gvg_mapper" output="screen" args=" ">
    <param name="laser_distance_x" value="$(arg laser_offset_x)" />
    <param name="laser_distance_y" value="$(arg laser_offset_y)" />
    <param name="leave_meetpoint_min_dist" value="$(arg leave_meetpoint_min_dist)" />
    <param name="leave_meetpoint_max_dist" value="$(arg leave_meetpoint_max_dist)" />
    <param name="relocalize_epsilon" type="double" value="0.1" />
    <param name="closest_distance_threshold" value="1.4" type="double" /> <!-- Distance from teh robot that hte meetpoints must be to allow it to be added: helps not add non-meetpoints -->
    <param name="exploration_policy" value="1" />    <!-- random walk = 0, ear-based = 1, user chosen = 2 -->
    <param name="vertex_matching_policy" value="1" />       <!-- strictly odometry based = 0, user confirmation = 1, vertex matching = 2 -->
    <param name="edge_length_threshold" value="1.5" type="double" />   <!-- cut-off threshold when comparing edge lengths (bearing matching) -->
    <param name="alwaysRelocalize" value="0" type="int" />   <!-- ask everytime=0, always relocalize=1, half the time relocalize=2 -->
  </node>


  <!--<node name="map_saver" pkg="gvg_mapper" type="MapSaver.py" output="screen">
    <param name="map_save_dir" value="map_log/" />
    <param name="map_save_update_rate" value="120" />
  </node>-->
</launch>
